### PR TITLE
PWGCF: Update MeanPtFlucIdentified.cxx

### PR DESCRIPTION
**Pull request by TGahlaut1**
Update PID cuts.

### CODE REVIEW

File: PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx [hunk 1]
```diff
@@ -22,6 +22,8 @@
 #include "Framework/ASoAHelpers.h"
 #include "Framework/HistogramRegistry.h"
 #include "Framework/HistogramSpec.h"
+#include "Framework/O2DatabasePDGPlugin.h"
+// #include "CCDB/BasicCCDBManager.h"
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:26]: Please, don't leave code commented

File: PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx [hunk 2]
```diff
@@ -43,58 +45,58 @@ double massPr = TDatabasePDG::Instance()->GetParticle(2212)->Mass();
...
+  Configurable<float> cfgCutNSigTpcEl{"cfgCutNSigTpcEl", 1.5, "TPC nSigma Electron veto cut"};
+  Configurable<float> cfgCutNSigTofEl{"cfgCutNSigTofEl", 1.5, "TOF nSigma Electron veto cut"};
+  Configurable<float> cfgCutNSig2{"cfgCutNSig2", 2.0, "nSigma cut (2)"};
+  Configurable<float> cfgCutNSig3{"cfgCutNSig3", 3.0, "nSigma cut (3)"};
+  Configurable<float> cfgCutPiPtMin{"cfgCutPiPtMin", 0.2, "Minimum pion p_{T} cut"};
+  Configurable<float> cfgCutKaPtMin{"cfgCutKaPtMin", 0.3, "Minimum kaon p_{T} cut"};
+  Configurable<float> cfgCutPrPtMin{"cfgCutPrPtMin", 0.5, "Minimum proton p_{T} cut"};
+  Configurable<float> cfgCutPiP1{"cfgCutPiP1", 0.65, "pion p cut-1"};
+  Configurable<float> cfgCutPiP2{"cfgCutPiP2", 0.75, "pion p cut-2"};
+  // Configurable<float> cfgCutPiP3{"cfgCutPiP3", 1.20, "pion p cut-3"};
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:67]: Same here

File: PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx [hunk 3]
```diff
@@ -192,108 +234,169 @@ struct meanPtFlucId {
...
       return false;
 
     if (!col.sel8())
       return false;
 
+    if (!col.selection_bit(o2::aod::evsel::kNoSameBunchPileup))
+      return false;
+
+    // if (!col.selection_bit(o2::aod::evsel::kNoITSROFrameBorder))
+    //   return false;
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:269]: Please, don't leave code commented

File: PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx [hunk 4]
```diff
@@ -316,184 +420,328 @@ struct meanPtFlucId {
...
+    hist.fill(HIST("QA/after/innerParam/h2_pvsm2"), track.mass() * track.mass(), track.tpcInnerParam());
+  }
+
   template <bool DataFlag, bool RecoFlag, typename T, typename U>
   void FillHistos(T const& col, U const& tracks)
   {
     int N_Pi = 0, N_Ka = 0, N_Pr = 0;
     int Nch = 0, NTPC = 0, N_FT0C = 0;
-    double Cent_FT0C = 0;
+    // double Cent_FT0C = 0;
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:641]: Same comment here

File: PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx [hunk 5]
```diff
@@ -316,184 +420,328 @@ struct meanPtFlucId {
...
-        hist.fill(HIST("QA/Proton/before/h2_TPCNsigma"), track.p(), track.tpcNSigmaPr());
-        hist.fill(HIST("QA/Proton/before/h2_TOFNsigma"), track.p(), track.tofNSigmaPr());
-        hist.fill(HIST("QA/Proton/before/h2_TpcTofNsigma"), track.tpcNSigmaPr(), track.tofNSigmaPr());
-        hist.fill(HIST("QA/Kaon/before/h2_TPCNsigma"), track.p(), track.tpcNSigmaKa());
-        hist.fill(HIST("QA/Kaon/before/h2_TOFNsigma"), track.p(), track.tofNSigmaKa());
-        hist.fill(HIST("QA/Kaon/before/h2_TpcTofNsigma"), track.tpcNSigmaKa(), track.tofNSigmaKa());
 
         // For Pions:
-        if (selPions(track)) {
+        if (selPions(track) == cfgSelPi /* && selPionsInnerParam(track) == cfgSelPiInnerParam */) {
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:671]: Same comment here

File: PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx [hunk 6]
```diff
@@ -316,184 +420,328 @@ struct meanPtFlucId {
...
-          hist.fill(HIST("QA/Pion/h2_TPCSignal"), track.p(), track.tpcSignal());
-          hist.fill(HIST("QA/Pion/h2_ExpTPCSignal"), track.p(), track.tpcExpSignalPi(track.tpcSignal()));
-          hist.fill(HIST("QA/after/h2_TOFSignal"), track.p(), track.beta());
-          hist.fill(HIST("QA/after/h2_TPCSignal"), track.p(), track.tpcSignal());
+          FillPionQAHistos(track);
         }
 
         // For Kaons:
-        if (selKaons(track)) {
+        if (selKaons(track) == cfgSelKa /* && selKaonsInnerParam(track) == cfgSelKaInnerParam */) {
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:679]: And here

File: PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx [hunk 7]
```diff
@@ -316,184 +420,328 @@ struct meanPtFlucId {
...
-          hist.fill(HIST("QA/Kaon/h2_TPCSignal"), track.p(), track.tpcSignal());
-          hist.fill(HIST("QA/Kaon/h2_ExpTPCSignal"), track.p(), track.tpcExpSignalKa(track.tpcSignal()));
-          hist.fill(HIST("QA/after/h2_TOFSignal"), track.p(), track.beta());
-          hist.fill(HIST("QA/after/h2_TPCSignal"), track.p(), track.tpcSignal());
+          FillKaonQAHistos(track);
         }
 
         // For Protons:
-        if (selProtons(track)) {
+        if (selProtons(track) == cfgSelPr /* && selProtonsInnerParam(track) == cfgSelPrInnerParam */) {
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:687]: And here

File: PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx [hunk 8]
```diff
@@ -316,184 +420,328 @@ struct meanPtFlucId {
...
-          if (std::abs(track.mcParticle().pdgCode()) == 321 && selKaons(track)) {
-            N_Ka++;
-            pt_Ka = track.pt();
-            moments(pt_Ka, &Q1_Ka, &Q2_Ka, &Q3_Ka, &Q4_Ka);
-            hist.fill(HIST("QA/Reco/Kaon/h_Pt"), track.pt());
-            hist.fill(HIST("QA/Reco/Kaon/h_Eta"), track.eta());
-            hist.fill(HIST("QA/Reco/Kaon/h_y"), track.rapidity(massKa));
-            hist.fill(HIST("QA/Reco/Kaon/h2_tpcSignal"), track.p(), track.tpcSignal());
-            hist.fill(HIST("QA/Reco/Kaon/h2_tofSignal"), track.p(), track.beta());
+          if (selKaons(track) == cfgSelKa /* || selKaonsInnerParam(track) == cfgSelKaInnerParam */) {
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:717]: And here

File: PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx [hunk 9]
```diff
@@ -316,184 +420,328 @@ struct meanPtFlucId {
...
+              FillProtonQAHistos(track);
+            }
           }
         }
       }
     }
 
     N_FT0C = col.multFT0C();
-    Cent_FT0C = col.centFT0C();
+    // Cent_FT0C = col.centFT0C();
```
Comment (victor-gonzalez) [inline PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:741]: !

**Review by victor-gonzalez — APPROVED** [review]
Please, consider my comments for the next iteration

I already mentioned before that there is a lot of code replicated and this is quite error prone and complicated to maintain. I would suggest to make a much more extensive use of the `template` mechanism to avoid as much as possible to replicate code

**Comment by TGahlaut1** [comment]
Thank you for the suggestions!
I am working on reducing code duplication by using templates more.

File: PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx [hunk 10]
```diff
@@ -728,46 +987,144 @@ struct meanPtFlucId {
...
+          hist.fill(HIST(dire[3]) + HIST("h_twopart_Mult_kurto"), NTPC, twopart_Pr, N_FT0C);
+        if (threepart_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_threepart_Mult_kurto"), NTPC, threepart_Pr, N_FT0C);
+        if (fourpart_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_fourpart_Mult_kurto"), NTPC, fourpart_Pr, N_FT0C);
       }
-      if (twopart_Pr != 0)
-        hist.fill(HIST(dire[3]) + HIST("h_twopart_Mult_var"), NTPC, twopart_Pr);
     }
   }
```
Comment (TGahlaut1) [inline PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:1126]: I was trying to use templates to reduce this histograms replication for different Identified particles in the code. but it wasn't working.

File: PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx [hunk 11]
```diff
@@ -728,46 +987,144 @@ struct meanPtFlucId {
...
+        if (twopart_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_twopart_Mult_kurto"), NTPC, twopart_Pr, N_FT0C);
+        if (threepart_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_threepart_Mult_kurto"), NTPC, threepart_Pr, N_FT0C);
+        if (fourpart_Pr != 0)
+          hist.fill(HIST(dire[3]) + HIST("h_fourpart_Mult_kurto"), NTPC, fourpart_Pr, N_FT0C);
       }
-      if (twopart_Pr != 0)
-        hist.fill(HIST(dire[3]) + HIST("h_twopart_Mult_var"), NTPC, twopart_Pr);
     }
```
Comment (TGahlaut1) [inline PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:1125]: I was trying to use templates to reduce this histograms replication for different Identified particles in the code. but it wasn't working.

Comment (victor-gonzalez) [reply PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:1125]: I'm confused!
What template parameter are you using here?

Comment (TGahlaut1) [reply PWGCF/EbyEFluctuations/Tasks/MeanPtFlucIdentified.cxx:1125]: I think I got it how to do this.
Initially I was doing some mistake in giving the directory (dire[]) name as argument.

**Pull request merged**